### PR TITLE
Deprecate the SonarQube msbuild tasks

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: Tasks/SonarQubePostTest/task.json [hunk 1]
```diff
@@ -2,17 +2,18 @@
     "id": "730D8DE1-7A4F-424C-9542-FE7CC02604EB",
     "name": "SonarQubePostTest",
     "friendlyName": "SonarQube for MSBuild - End Analysis",
-    "description": "Finish the analysis and upload the results to SonarQube",
+    "description": "[DEPRECATED] Finish the analysis and upload the results to SonarQube",
     "helpMarkDown": "[More Information](https://go.microsoft.com/fwlink/?LinkId=620063)",
     "category": "Build",
+    "deprecated": true,
```
Comment (bgavrilMS) [inline Tasks/SonarQubePostTest/task.json]: @bryanmacfarlane - marking these as deprecated as discussed.

File: Tasks/SonarQubePostTest/SonarQubePostTest.ps1 [hunk 2]
```diff
@@ -2,6 +2,10 @@ Write-Verbose "Starting SonarQube PostBuild Step"
 
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 
+Write-Warning "We are transferring ownership of the SonarQube related build tasks to Sonar Source. Please replace these build tasks with the ones provided by Sonar Source.
```
Comment (jmprieur) [inline Tasks/SonarQubePostTest/SonarQubePostTest.ps1]: I'd prefer. "The ownership of the SonarQube related build tasks is transferred to SonarSource". Also SonarSource is in one word. We'd need to have the link on their extension, also. Maybe we should involve Freddy / Tamas in this PR?

File: Tasks/SonarQubePostTest/SonarQubePostTest.ps1 [hunk 3]
```diff
@@ -2,6 +2,10 @@ Write-Verbose "Starting SonarQube PostBuild Step"
 
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 
+Write-Warning "We are transferring ownership of the SonarQube related build tasks to Sonar Source. Please replace these build tasks with the ones provided by Sonar Source. 
+To do this, visit the marketplace to install the extension created by Sonar Source: https://aka.ms/sqextension
```
Comment (jmprieur) [inline Tasks/SonarQubePostTest/SonarQubePostTest.ps1]: SonarSource (1 word)

File: Tasks/SonarQubePreBuild/SonarQubePreBuild.ps1 [hunk 4]
```diff
@@ -27,6 +27,10 @@ Write-Verbose "includeFullReport = $includeFullReport"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 
+Write-Warning "We are transferring ownership of the SonarQube related build tasks to Sonar Source. Please replace these build tasks with the ones provided by Sonar Source.
```
Comment (jmprieur) [inline Tasks/SonarQubePreBuild/SonarQubePreBuild.ps1]: Same message: "Ownership of the SonarQube for MSBuild tasks is transfered to SonarSource. Please replace these build tasks by the one brought by SonarSource's extension on the marketplace".

**Comment by HSAR** [comment]
Save for "Sonar Source" => "SonarSource" comments left by @jmprieur, LGTM.

File: Tasks/SonarQubePostTest/SonarQubePostTest.ps1 (hunk 2 above)
Comment (bgavrilMS) [inline Tasks/SonarQubePostTest/SonarQubePostTest.ps1]: There are aka.ms links in the messge. I'll update them once the final urls become
 available.

**Comment by HSAR** [comment]
LGTM

**Comment by psyvision** [comment]
So what happens now?  When will the links sqdeprecated and sqtransfer links become active along with the new plugin in the marketplace?

Are there additional items being released along with the 6.2 release of SonarQube?

**Comment by bgavrilMS** [comment]
Hi @psyvision - SonarSource are still testing their extension. Once they publish their extension I'll update those URLs to point to it and to a blog post explaining more. 

SonarSource keep their backlog in the public, so have a look for yourself - https://jira.sonarsource.com/browse/TBSTFNP/?selectedTab=com.atlassian.jira.jira-projects-plugin:roadmap-panel

There will be an aditional SonarQube task that wrapps the SonarQube Scanner, enabling scenarios such as scanning non-C# / non-Java payloads.

**Comment by psyvision** [comment]
Many thanks @bgavrilMS - I look at their JIRA occasionally to do with bug fixes (like Active Directory/SSO integration) but find it hard to navigate at times.

**Pull request merged**